src/common/util.cpp

```diff
@@ -79,3 +76,36 @@ std::vector<uint8_t> hexToBytes(std::string hexStr) {
 
     return bytes;
 }
+
+const int compareFloats(const float a, const float b, const size_t epsilon) {
+    /*
+    A simple decimal comparison might involve subtraction, fabs, and the result being less than a smaller decimal.
+    This method converts the IEEE float values into lexicographically-ordered two's-complement integers.
+    If (float)a > (float)b then (int)a > (int)b; but, precision concerns are lessened.
+    The error becomes a count of "the maximum number of lexico-numbers allowed between equal lexico-a and lexico-b."
+    To avoid encountering a NaN value, error can not risk being too big, however.
+    */
+    if(epsilon >= 4194304) {
+        throw std::invalid_argument("float comparison - error margin too big");
+    }
+
+	int epsilon2 = *(int*)&epsilon;
```

Comment:
No need for this as far as I can tell - make your epsilon parameter an int instead of a size_t and you should be able to get rid of epsilon2.